src/plugins/npm/npm.ts

```diff
@@ -0,0 +1,80 @@
+import * as fs from 'fs';
+import { promisify } from 'util';
+
+import { ILogger } from '../../github-release';
+import { IAutoHooks, IPlugin } from '../../main';
+import SEMVER from '../../semver';
+import execPromise from '../../utils/exec-promise';
+import getConfigFromPackageJson from './package-config';
+
+const readFile = promisify(fs.readFile);
+
+function isMonorepo() {
+  return fs.existsSync('lerna.json');
+}
+
+export default class NPMPlugin implements IPlugin {
+  public name = 'NPM';
+
+  public apply(auto: IAutoHooks, logger: ILogger) {
```

Comment:
The arg to `apply` is typically an instance of `auto` rather than the hooks object. 

So you'd use 

```js
public apply(auto: AutoRelease) {
  auto.logger.log()
  auto.hooks.getAuthor.tapPromise ...
}
```